templates/plumbing/applications.yaml

```
@@ -1,5 +1,6 @@
 {{- $namespace := print $.Values.global.pattern "-" $.Values.clusterGroup.name }}
 {{- range .Values.clusterGroup.applications }}
+{{- $projectName := .argoProject | default (.project | default "default") -}} {{- /* Read argoProject if set, otherwise .projects and otherwise set it to "default" */}}
```

We use `coalesce` to do similar things in this chart so maybe it's a cleaner fit here as well. (Also, tiny typo in the comment with `.projects`.)

```suggestion
{{- $projectName := coalesce .argoProject .project "default" -}} {{- /* Read argoProject if set, otherwise .project and otherwise set it to "default" */}}
```